### DISCUSSION TITLE
Vectored images take too long to cut... and add Hundreds of lines

### DISCUSSION THREAD

**Discussion by JordiJTDesign** [arkypita/LaserGRBL#1726] [Q&A]
I'm struggling with too long times for cutting wood!

Here's the problem:
Recently I've been using "LaserGRBL" and I'm still learning how it works.
Until now I've used my own designs (made in Illustrator, save as JPG) to use the "Vectorize![Experimental]" tool after that to CUT with the laser cutting  (V.200 / SMax 1.000 / Repeat x15) --> With a DM wood of 2mm

![A_Tall](https://user-images.githubusercontent.com/97853748/149677881-8d2328e9-3460-41b7-8ced-20ab9ff184e3.jpg)

However, for the next design (also made in Illustrator, save as JPG), the time for cutting it is TOO high (30 minutes per time, 15 times = 9 hours)

![image](https://user-images.githubusercontent.com/97853748/149677909-358657ea-f4bc-4f16-92a8-b8151d4efb32.png)

So now I've tried to use an SVG (from Illustrator too) with all shapes together and without overlapped lines.
And sadly, guess what, this just increased the time: the triple amount of lines and time

![image](https://user-images.githubusercontent.com/97853748/149677989-64cded3a-400d-4a98-b4b0-42ee6a8b99e7.png)

What can I do to solve it? Thank you so much for your time and attention!

Note: My LaserCutting is an Atomstack A5

**Comment by arkypita** [comment]
You are doing a design that is very big: 800 x 600 mm
At the speed you set it is normal that the estimated time is so high

**Reply by JordiJTDesign** [reply]
Thank you so much, you were right!

When I create the SVG through Illustrator, it doesn't respect the measures and it multiplies them by x4 for no apparent reason... then, once I created something in Illustrator, I put it on InkScape and then I get the right measurements.

I've also tried to increase the speed to 400 but I have problems with the cut: it doesn't go through 2mm of DM wood.

Note: And last, I have a problem with layouts: I don't understand why but it doesn't follow a logic sequence at cutting, it goes from one corner to another instead of going to the closest lines. Why do you think this happens?

**Comment by ttjenik** [comment]
Hallo,
I have a similar problem with importing SVG vector curves from Inkscape.
I can't or don't know how to set the optimal cutting path in the program so that the whole one contour is cut at once.
In the picture you can see how the path is interrupt and the laser jump from one contour to another.
Is there some settings: "Contour optimization" or "Cut contour in one cut"? 
![obrazek](https://user-images.githubusercontent.com/98387436/150952736-a0b71e67-d944-47e4-8118-e06be9a364c8.png)
Thank you for your answer
Jan

**Reply by Tribo-2dot0** [reply]
If in inkscape you put each shape on a separate layer it will cut in order, top layer first and bottom layer last.

**Comment by ttjenik** [comment]
Thank you for the advice.

I don't know if I understood your answer well, but you thought of arranging the lines in the correct order.
OK, I tried your advice on this drawing. Heart (downloaded from the Internet)

In the drawing without modifications, the laser moved as shown on the left screen.
Then I arranged them in the optimal order.
The picture on the right shows that the order has changed, but the cut is still interrupted.
![obrazek](https://user-images.githubusercontent.com/98387436/151697071-7bbad5eb-f443-489a-9580-1f6d38e529a5.png)
I tried a different order of lines and the result is similar. 
![obrazek](https://user-images.githubusercontent.com/98387436/151697759-d2a74606-d0b1-455b-9c3e-687f79fa12b7.png)

I see the problem is that every single line starts on the right side of the line. The program is simply unable to evaluate the current position and the nearest point of the next curve to optimize the start. 
Maybe I just have the wrong settings. So I ask again. Is there a way to set the cut path optimization? 

Thanks Jan
